web/js/date/timeline-input.js

```diff
@@ -10,177 +11,297 @@ import lodashParseInt from 'lodash/parseInt';
 export function timelineInput(models, config, ui) {
   var tl = ui.timeline;
   var model = models.date;
-  var timer;
   var self = {};
   var rollingDate;
   self.direction = 'forward';
   self.interval = 'day';
   self.delta = 1;
   self.active = false;
   self.delay = 500;
+  var animator = null;
+  var keyDown;
 
   var $incrementBtn = $('#right-arrow-group');
   var $decrementBtn = $('#left-arrow-group');
 
-  var forwardNextMinute = function () { // FIXME: Limit animation correctly
-    self.delta = 10;
-    var nextMinute = new Date(new Date(model.selected)
-      .setUTCMinutes(model.selected.getUTCMinutes() + 10));
-    if (nextMinute <= util.now()) {
-      animateForward('minute', 10);
-    } else {
-      self.stop();
-    }
-  };
+  var init = function () {
+    $incrementBtn
+      .mousedown(function (e) {
+        e.preventDefault();
+        switch (ui.timeline.config.currentZoom) {
+          case 1:
+            animateByIncrement(1, 'year');
+            break;
+          case 2:
+            animateByIncrement(1, 'month');
+            break;
+          case 3:
+            animateByIncrement(1, 'day');
+            break;
+          case 4:
+            animateByIncrement(10, 'minute');
+            break;
+          default:
+            animateByIncrement(1, 'day');
+        }
+      })
+      .mouseup(stopper);
 
-  var forwardNextDay = function () { // FIXME: Limit animation correctly
-    self.delta = 1;
-    var nextDay = new Date(new Date(model.selected)
-      .setUTCDate(model.selected.getUTCDate() + 1));
-    if (nextDay <= util.today()) {
-      animateForward('day', 1);
-    } else {
-      self.stop();
-    }
-  };
+    $decrementBtn
+      .mousedown(function (e) {
+        e.preventDefault();
+        switch (ui.timeline.config.currentZoom) {
+          case 1:
+            animateByIncrement(-1, 'year');
+            break;
+          case 2:
+            animateByIncrement(-1, 'month');
+            break;
+          case 3:
+            animateByIncrement(-1, 'day');
+            break;
+          case 4:
+            animateByIncrement(-10, 'minute');
+            break;
+          default:
+            animateByIncrement(-1, 'day');
+        }
+      })
+      .mouseup(stopper);
 
-  var forwardNextMonth = function () {
-    self.delta = 1;
-    var nextMonth = new Date(new Date(model.selected)
-      .setUTCMonth(model.selected.getUTCMonth() + 1));
-    if (nextMonth <= util.today()) {
-      animateForward('month', 1);
-    } else {
-      self.stop();
-    }
-  };
+    $(document)
+      .mouseout(stopper)
+      .keydown(function (event) {
+        if (event.target.nodeName === 'INPUT' || keyDown === event.keyCode) return;
+        switch (event.keyCode) {
+          case util.key.LEFT:
+            if (models.date.selectedZoom === 4) {
```

Comment:
This section of the code looks like it needs some latest changes from develop. There was an addition added to v2.3.0 which changed the left / right arrow functionality to reflect the current zoom level.
```
switch (models.date.selectedZoom) {
      case 1:
        animateReverse('year', -1);
        break;
      case 2:
        animateReverse('month', -1);
        break;
      case 3:
        animateReverse('day', -1);
        break;
      case 4:
        animateReverse('minute', -10);
        break;
}
```

Reply:
Good find!